src/components/Widgets.tsx

```diff
@@ -94,70 +78,56 @@ export class UploadInput extends React.Component<{
     this.inputElement.click();
   }
   render() {
-    return <input id="file-upload-input" ref={ref => this.setInputElement(ref)} type="file" onChange={this.props.onChange} multiple hidden/>;
+    return <input id="file-upload-input" ref={ref => this.setInputElement(ref)} type="file" onChange={this.props.onChange} multiple hidden />;
   }
 }
 
-export class ListItem extends React.Component<{
+export function ListItem(props: {
   label: string;
   onClick?: Function;
   icon?: string;
   selected?: boolean;
   value: any;
   error?: string;
-}, {
-
-  }> {
-
-  render() {
-    let className = "list-item";
-    if (this.props.selected) {
-      className += " selected";
-    }
-    let content = <div className="label">{this.props.label}</div>;
-    if (this.props.error) {
-        content = <div className="list-item-flex">
-          <div className="label">{this.props.label}</div>
-          <div className="error">{this.props.error}</div>
-        </div>;
-    }
-    return <div className={className} onClick={this.props.onClick as any}>
-      <div className={"monaco-icon-label file-icon " + this.props.icon} />
-      {content}
+}) {
+  let className = "list-item";
+  if (props.selected) {
```

Comment:
You could use `selected`, `label`, `error` without `props.`

Reply:
Hmmm, it didn't work here, can you recheck?

Reply:
Sorry. It should be declare like this:
```ts
export function ListItem({
  label,
  onClick,
  icon,
  selected,
  value,
  error
}: {
  label: string;
  onClick?: Function;
  icon?: string;
  selected?: boolean;
  value: any;
  error?: string;
}) {
}
```

Reply:
Interesting syntax, thanks. It uses however two lines for each property, isn't better to limit that when we have fewer number of parameters?

Reply:
Definitely, you could use just minimal set of properties which you would use inside render function. Actually this destruction syntax. I the same as:
```ts
export function ListItem(props: { label: string; onClick?: Function }) {
   const { label, onClick } = props;
}
```

Reply:
All done, have a look :)